woocommerce/woocommerce
# Update Action Scheduler to v2.1.0

### All Submissions:


### Changes proposed in this Pull Request:

This PR updates Action Scheduler to v2.1.0. This is primarily to follow up on the request to change Action Scheduler's function prefixes from `wc_` to `as_`. Original request for the prefix change: https://github.com/woocommerce/woocommerce/pull/20030#pullrequestreview-139753264

For the specific changeset for this, see https://github.com/Prospress/action-scheduler/pull/181

It also adds a new `WC_Queue_Interface::cancel_all()` method to help address some of the ambiguity with the existing `WC_Queue_Interface::cancel()` method (docblocks also updated). This came up previously here: https://github.com/woocommerce/woocommerce/pull/20030#issuecomment-408043930

For the specific changeset for this, see https://github.com/Prospress/action-scheduler/pull/201/

In addition to that, v2.1.0 fixes a rare timezone issue that was already being handled by WC since v3.0, so was likely encountered previously, and a few smaller issues. For details, see: https://github.com/Prospress/action-scheduler/milestone/3?closed=1

This should be the final piece of #18003.

Beyond the code, I'd also like to publish some docs, probably in the Developer Wiki, on how the job queue APIs can be used via 3rd party code. Please let me know if you'd like to proceed with those.

### How to test the changes in this Pull Request:

1. Checkout the branch
2. Make sure you have a webhook enabled
3. Run through the steps to trigger that webhook
4. Confirm the webhook was sent to the `Delivery URL`
5. Find the history of the webhook in [Action Scheduler's admin view](https://github.com/Prospress/action-scheduler/#managing-scheduled-actions)

### Other information:

* [x] Have you written new tests for your changes, as applicable? (New tests in Action Scheduler for `cancel_all()` method)

### Changelog entry

This changelog entries can be used for the changes added here and the initial addition of AS in #20030, which didn't include changelog entries: 

```
* Dev - Add an extensible/swappable job queue via WC_Action_Queue and WC_Queue_Interface. #20030
* Dev - Use WC_Action_Queue for schedule and delivery of webhooks. #20030
* Dev - Include Action Scheduler v2.1.0 and use it for the default job queue. #21424 /  #20030
```
